library/psa_crypto.c

```diff
@@ -4717,6 +4717,9 @@ psa_status_t psa_aead_set_lengths(psa_aead_operation_t *operation,
         goto exit;
     }
 
+    /* Don't check operation->nonce_set here because Mbed TLS supports
+     * psa_aead_set_lengths() after setting the nonce for backward
+     * compatibility. */
```

Comment:
How will we remember to change this when we remove deprecations? Could we add the check here gated with `MBEDTLS_DEPRECATED_REMOVED`?